mlir/include/mlir/Dialect/OpenMP/Passes.td

```diff
@@ -0,0 +1,54 @@
+//===-- Passes.td - OpenMP pass definition file -------------*- tablegen -*-===//
```

Comment:
Previous OpenMP passes were added in https://github.com/llvm/llvm-project/blob/main/flang/include/flang/Optimizer/Transforms/Passes.td#L321.

Even though I agree that OpenMP passes like this one and the ones linked above can be separated into their own file since they are not flang specific, but I think for consistency it would be better to use the already existing file to have a more centralized view of all OpenMP-related passes.

Reply:
Just a little background on why the passes are there from what I recall at least (more for information to help make informed decisions or for someone else to chime in with a preference), the `OMPMarkDeclareTargetPass`/`OMPFunctionFiltering `are there as they were initially in an experimental phase (alongside another now erased pass) to verify if we'd keep them in the offloading flow and see how they'd interact with each other overtime as they are interlinked behavior. Not sure when they'd best to be moved, it's coming up to their anniversary at this point, but it's something that can wait in any case.

The `OMPDescriptorMapInfoGenPass `is very much Fortran specific at the moment, it primarily handles FIR `BoxType`'s, so it's not feasible at the moment (or perhaps ever) to move to the OpenMP dialect as it has FIR dependencies and FIR is an external dialect to the MLIR project.

Reply:
Thanks for the info @agozillon. @ergawy  - Like you said, I kept this in mlir only because it isnt really flang specific and I anticipated that if I put it in flang someone might call me out for it. OTOH keeping it in flang would have made testcase generation easier as I'd have been able to load the `fir` dialect into the MLIR context. Now I had to rely on ``fir-opt`` to convert all ``fir`` ops to ``llvm`` dialect ops, but  `fir-opt` is broken in a weird way such that I couldn't run ``fir-to-llvm-ir`` lowering to rid my testcases of ``fir`` ops. (Aside: I hope to find some time to chase down the `fir-opt` problem)